### ISSUE TITLE
Using it for Multilabel classification? 

### ISSUE THREAD

**Issue by enod** [bentrevett/pytorch-sentiment-analysis#3]
Thanks for the awesome tutorial.

4 - Convolutional Sentiment Analysis.ipynb
Do you think it is possible to use this part of tutorial in multilabel classification case? (eg: kaggle toxic comment) 
It seems by changing output dimension to 6 is not working. Or do I have to use nn.softmax with BCEloss()?

**Comment by greed2411** [comment]
For multi label classification one is supposed to use `sigmoid` over `softmax`, because softmax makes sure the output of all values add up to zero. Therefore, you can get it to predict only one value. Whereas sigmoid there is no such restriction.

**Comment by enod** [comment]
@greed2411 
Thanks for the prompt answer. So in that case, does model should look like following?
```

class CNN(nn.Module):
    def __init__(self, vocab_size, embedding_dim, n_filters, filter_sizes, output_dim, dropout):
        super().__init__()
        self.embedding = nn.Embedding(vocab_size, embedding_dim)
        self.conv_0 = nn.Conv2d(in_channels=1, out_channels=n_filters, kernel_size=(filter_sizes[0],embedding_dim))
        self.conv_1 = nn.Conv2d(in_channels=1, out_channels=n_filters, kernel_size=(filter_sizes[1],embedding_dim))
        self.conv_2 = nn.Conv2d(in_channels=1, out_channels=n_filters, kernel_size=(filter_sizes[2],embedding_dim))
        self.fc = nn.Linear(len(filter_sizes)*n_filters, output_dim)
        self.dropout = nn.Dropout(dropout)
        
    def forward(self, x):
        
        #x = [sent len, batch size]
        
        x = x.permute(1, 0)
                
        #x = [batch size, sent len]
        
        embedded = self.embedding(x)
                
        #embedded = [batch size, sent len, emb dim]
        
        embedded = embedded.unsqueeze(1)
        
        #embedded = [batch size, 1, sent len, emb dim]
        
        conved_0 = F.relu(self.conv_0(embedded).squeeze(3))
        conved_1 = F.relu(self.conv_1(embedded).squeeze(3))
        conved_2 = F.relu(self.conv_2(embedded).squeeze(3))
            
        #conv_n = [batch size, n_filters, sent len - filter_sizes[n]]
        
        pooled_0 = F.max_pool1d(conved_0, conved_0.shape[2]).squeeze(2)
        pooled_1 = F.max_pool1d(conved_1, conved_1.shape[2]).squeeze(2)
        pooled_2 = F.max_pool1d(conved_2, conved_2.shape[2]).squeeze(2)
        
        #pooled_n = [batch size, n_filters]
        
        cat = self.dropout(torch.cat((pooled_0, pooled_1, pooled_2), dim=1))

        #cat = [batch size, n_filters * len(filter_sizes)]
            
        return F.sigmoid(self.fc(cat))
```

**Comment by greed2411** [comment]
Yes that's how it should be. 

On another case, can't think of a reason why @bentrevett used `Conv2d` instead of `Conv1d` for text data.

**Comment by bentrevett** [comment]
@greed2411, I'm not sure why I used it either. I believe I found it found it easier to get my head around when thinking about it as 2d. 

I'm currently in the process of updating these tutorials to TorchText 0.3 as it has better integration with PyTorch 0.4 and will change to a `Conv1d`.
